### PR TITLE
Small cleanup, add some new features

### PR DESCRIPTION
**Pull request by johnbieren**
Rundown:

- Added the ability to pass in a koji task id. The default is null, but if one is passed in, it will bypass submitting a build to koji and instead download the task's artifacts and proceed with those in the workspace
- I like the for each with the library checkout. However, the library based on the repo where this code lives (so right now it would be the joejstuart library) needs to be checked out a little differently, as that depends on the ghprb commit and stuff. I did not patch this part
- Added a repoquery script in the rpmbuild stage because I want to verify we have a working repo to work with before proceeding.
- Removed some legacy stuff you didn't need
- The ostree-boot-image doesn't currently do anything since we don't have $expected, so turn that off for now
- If no tests exist, skip that entire stage
- Remove duplication and unneeded code from the library hosted here

Note that this stuff is dependent on my next PR to ci-pipeline, which I will comment here with a link to. It will be adding the repoquery and pull_old_task scripts

Signed-off-by: Johnny Bieren <jbieren@redhat.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -280,27 +285,33 @@ podTemplate(name: podName,
                     }
                 }
 
-                currentStage = "fedora-package-pipeline-functional-tests"
-                stage(currentStage) {
+                // Only run this stage if tests exist
+                if (pipelineUtils.checkTests(env.fed_repo)) {
```
Comment (joejstuart) [inline Jenkinsfile]: I put this check in JenkinsfileTrigger also. Should it be taken out of there? Trigger would run on all messages then.

Comment (johnbieren) [reply Jenkinsfile]: Oh, did you? I must have missed it. Did you use the checkTests function? Can you link where?

I am not sure what we are supposed to do. I am going to ask Thursday. I am not sure if they want us to build + compose an image for packages with no tests or not. I see value in knowing if the package can build and compose a host whether or not there are tests after, obviously the tests are a bonus, but that is just me. I don't speak for the others, so I gotta get a straight answer from them.

Comment (joejstuart) [reply Jenkinsfile]: Yeah. It's commented out. https://github.com/joejstuart/upstream-fedora-pipeline/blob/master/JenkinsfileTrigger#L56

I agree with you on building for every package, but let's see.

Comment (johnbieren) [reply Jenkinsfile]: Ah, I see. So the feedback I got from Miroslav was "I don't want a green passed stage if no tests ran." So I wrote what I proposed here and is merged in the downstream pipeline. Then the discussion started of "should we even do anything if tests don't exist?" which has yet to be resolved. /me shrugs

**Pull request merged**